app/src/main/java/org/sagebionetworks/research/mpower/logging/LoggingViewModel.java

```diff
@@ -0,0 +1,138 @@
+package org.sagebionetworks.research.mpower.logging;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import android.support.annotation.AnyThread;
+import android.support.annotation.MainThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import org.joda.time.DateTime;
+import org.sagebionetworks.bridge.android.manager.ActivityManager;
+import org.sagebionetworks.bridge.rest.model.ScheduledActivity;
+import org.sagebionetworks.research.mpower.TaskLauncher;
+import org.sagebionetworks.research.mpower.TaskLauncher.TaskLaunchState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * The ViewModel forms the presentation layer (ViewModel or Presenter). It is responsible for receiving actions from
+ * the views, communicating with the data layer, applying UI logic, keeping state, and exposing relevant data to the
+ * UI for rendering. It encapsulates logic and state while not depending on the Android framework, thus making it
+ * suitable for JVM unit testing.
+ * <p>
+ * https://medium.com/upday-devs/android-architecture-patterns-part-3-model-view-viewmodel-e7eeee76b73b
+ * <p>
+ * https://medium.com/@rohitsingh14101992/lets-keep-activity-dumb-using-livedata-53468ed0dc1f
+ */
+public class LoggingViewModel extends ViewModel {

```

Comment:
How will we be approaching Kotlin vs Java for the Data layer?  My vote would be to do it in Kotlin since it reads more like Swift and in the end will probably look more like the DataSource classes in the iOS Research framework.

Reply:
I'm on board with moving to Kotlin, we have some open questions so SageResearch has only a bit of Kotlin in the data layer. 

There's also [Kotlin Multiplatform](https://kotlinlang.org/docs/reference/multiplatform.html) which currently supports Java and JS, and [Kotlin/Native](https://kotlinlang.org/docs/reference/native-overview.html) is in development preview and supports iOS, AppleOS and other platforms. It'd be great to share data/models and other business logic across platform.

Reply:
I'm also on board with this being written in Kotlin.

Reply:
I'm cool with whichever you guys feel is the best approach. 

That said, with Apple-OS, we took the approach of *only* using Obj-C where there isn't support in Swift. (Formatters, Exception Handling) We did so b/c it is easier to maintain code when you are only working in one language.

So, is Kotlin support there yet? (Swift is on v4.2 and it took a while for it to really be viable)

How much could we leverage the Swift code already written verse just making the work to refactor and maintain more time-consuming?